src/components/secret-display/SecretDisplaySection.tsx

```diff
@@ -0,0 +1,181 @@
+  color: colors.appleBlue,
+  name: 'copy',
+}))({
+  ...position.sizeAsObject(16),
+  marginTop: 0.5,
+});
+
+const ToggleSecretButton = styled(Button)(({ theme: { colors } }: any) => ({
+  ...margin.object(0, 20),
+  ...shadow.buildAsObject(0, 5, 15, colors.purple, 0.3),
+  backgroundColor: colors.appleBlue,
+}));
+
+const LoadingSpinner = android ? Spinner : ActivityIndicator;
+
+interface SecretDisplaySectionProps {
+  onSecretLoaded?: (seedExists: boolean) => void;
+  onWalletTypeIdentified?: (walletType: EthereumWalletType) => void;
+}
+
+export default function SecretDisplaySection({
+  onSecretLoaded,
+  onWalletTypeIdentified,
+}: SecretDisplaySectionProps) {
+  const { params } = useRoute();
+  const { selectedWallet, wallets } = useWallets();
+  const walletId = (params as any)?.walletId || selectedWallet.id;
+  const currentWallet = wallets?.[walletId];
+  const [visible, setVisible] = useState(true);
+  const [isRecoveryPhraseVisible, setIsRecoveryPhraseVisible] = useState(false);
+  const [seed, setSeed] = useState<string | null>(null);
+  const [type, setType] = useState(currentWallet?.type);
+
+  const loadSeed = useCallback(async () => {
+    try {
+      const s = await loadSeedPhraseAndMigrateIfNeeded(walletId);
+      if (s) {
+        const walletType = identifyWalletType(s);
+        setType(walletType);
+        onWalletTypeIdentified?.(walletType);
+        setSeed(s);
+      }
+      setVisible(!!s);
+      onSecretLoaded?.(!!s);
+      setIsRecoveryPhraseVisible(!!s);
+    } catch (e: any) {
+      logger.sentry('Error while trying to reveal secret', e);
+      if (e?.message === createdWithBiometricError) {
+        setIsRecoveryPhraseVisible(false);
+      }
+      captureException(e);
+      setVisible(false);
+      onSecretLoaded?.(false);
+    }
+  }, [onSecretLoaded, onWalletTypeIdentified, walletId]);
+
+  useEffect(() => {
+    // Android doesn't like to show the faceID prompt
+    // while the view isn't fully visible
+    // so we have to add a timeout to prevent the app from freezing
+    android
+      ? setTimeout(() => {
+          loadSeed();
+        }, 300)
+      : loadSeed();
+  }, [loadSeed]);
+
+  const typeLabel = type === WalletTypes.privateKey ? 'key' : 'phrase';
+
+  const { colors } = useTheme();
+
+  const renderStepNoSeeds = useCallback(() => {
+    if (isRecoveryPhraseVisible) {
+      return (
+        <Box
+          alignItems="center"
+          justifyContent="center"
+          paddingHorizontal="60px"
+        >
+          <Stack space="10px">
+            <Text align="center" color="secondary" size="18px" weight="regular">
+              {lang.t('back_up.secret.you_need_to_authenticate', {
+                typeName: typeLabel,
+              })}
+            </Text>
+            <ToggleSecretButton onPress={loadSeed}>
+              {/* @ts-ignore */}
+              <BiometricButtonContent
+                color={colors.white}
```

Comment:
On dark mode the "View secret phase button is white"

Reply:
was poking through the code trying to find the issue. Was wondering if this was it lol

Reply:
@BrodyHughes i can fix it in https://github.com/rainbow-me/rainbow/pull/4051

Reply:
Ohh ok. I just QA'd that and it looks good aside from this regression.

Reply:
@pugson what do you think about including this? if not, nbd i got it 

https://linear.app/rainbow/issue/TEAM1-133/if-you-try-to-view-seed-phrase-and-decline-biometrics-our-explainer

Reply:
@benisgold sure thing i got it